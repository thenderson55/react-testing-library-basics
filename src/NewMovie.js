import React from 'react'
import MovieForm from './MovieForm'

function NewMovie() {
  return (
    <div data-testid="component-new-movie">
      <h2 data-testid="page-title">New Movie</h2>
      <MovieForm/>
    </div>
  )
}

export default NewMovie
