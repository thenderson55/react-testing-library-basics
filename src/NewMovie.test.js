import React from "react";
import { render, fireEvent, cleanup } from "@testing-library/react";
import NewMovie from "./NewMovie";

afterEach(cleanup);

test('should render NewMovie without error', () => {
  const { getByTestId, debug, queryByTestId, container, getByText } = render(<NewMovie/>)
  debug()

  const component = getByTestId('component-new-movie')
  expect(component).toBeTruthy()

  const title = getByTestId('page-title')
  expect(title.textContent).toBe('New Movie')

  expect(queryByTestId('movie-form')).toBeTruthy()

  // expect(container.firstChild).toMatchSnapshot()

  // fireEvent.click(getByText('Submit'))
});