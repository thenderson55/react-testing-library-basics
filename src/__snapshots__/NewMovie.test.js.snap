// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NewMovie without error 1`] = `
<div
  data-testid="component-new-movie"
>
  <h2
    data-testid="page-title"
  >
    New Movie
  </h2>
  <div>
    <form
      data-testid="movie-form"
    >
      <label
        for="test"
      >
        <input
          id="text"
          type="text"
        />
      </label>
      <button>
        Submit
      </button>
    </form>
  </div>
</div>
`;
